generator/internal/sample_generator.cc

```diff
@@ -124,7 +93,71 @@ void SetClientEndpoint(std::vector<std::string> const& argv) {
   HeaderPrint(R"""(options));
   //! [set-client-endpoint]
 }
+)""");
+}
+
+void SampleGenerator::GenerateSetClientUniverseDomainSample() {
+  HeaderPrint(R"""(
+void SetClientUniverseDomain(std::vector<std::string> const& argv) {
+  if (!argv.empty()) {
+    throw google::cloud::testing_util::Usage{"set-client-universe-domain"};
+  }
+  //! [set-client-universe-domain]
+  google::cloud::Options options;
+
+  // AddUniverseDomainOption interrogates the UnifiedCredentialsOption, if set,
+  // in the provided Options for the Universe Domain associated with the
+  // credentials and adds it to the set of Options.
+  // If no UnifiedCredentialsOption is set, GoogleDefaultCredentials are used.
+  ud_options = google::cloud::AddUniverseDomainOption(std::move(options));
```

Comment:
Is `ud_options` assumed to be present elsewhere in the generated samples? I'm asking cause I would expect a local declaration with `auto`.